src/js/components/sharedComponents/checkbox/PrimaryCheckboxType.jsx

```diff
@@ -0,0 +1,188 @@
+/**
+ * PrimaryCheckboxType.jsx
+ * Created by michaelbray on 5/17/17.
+ */
+
+import React from 'react';
+
+import { recipientTypes } from 'dataMapping/search/recipientType';
+import { Set } from 'immutable';
+
+import SecondaryCheckboxType from './SecondaryCheckboxType';
+import CollapsedCheckboxType from './CollapsedCheckboxType';
+import SingleCheckboxType from './SingleCheckboxType';
+
+const propTypes = {
+    id: React.PropTypes.string,
+    name: React.PropTypes.string,
+    filters: React.PropTypes.array,
+    bulkRecipientTypeChange: React.PropTypes.func,
+    value: React.PropTypes.string,
+    filterType: React.PropTypes.string,
+    selectedCheckboxes: React.PropTypes.object
+};
+
+const defaultProps = {
+    id: '',
+    name: '',
+    filters: [],
+    value: '',
+    filterType: '',
+    selectedCheckboxes: new Set()
+};
+
+const ga = require('react-ga');
+
+export default class PrimaryCheckboxType extends React.Component {
+    static logPrimaryTypeFilterEvent(type, filter) {
+        ga.event({
+            category: 'Search Page Filter Applied',
+            action: `Selected ${filter} Type`,
+            label: type
+        });
+    }
+
+    static logDeselectFilterEvent(type, filter) {
+        ga.event({
+            category: 'Search Page Filter Applied',
+            action: `Deselected ${filter} Type Children`,
+            label: type
+        });
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showSubItems: false,
+            arrowState: 'collapsed',
+            selectedChildren: false,
+            allChildren: false
+        };
+
+        // bind functions
+        this.toggleSubItems = this.toggleSubItems.bind(this);
+        this.toggleChildren = this.toggleChildren.bind(this);
+    }
+
+    componentDidMount() {
+        this.compareFiltersToChildren();
+    }
+
+    componentDidUpdate(prevProps) {
```

Comment:
I've been trying to move things over to `componentWillReceiveProps` when possible. This event is where [Facebook recommends](https://facebook.github.io/react/docs/react-component.html#componentwillreceiveprops) you put logic that sets states based on incoming props. Same goes for `componentWillMount` (as appropriate; you'll use `this.props` instead of `nextProps` though).